### PR TITLE
Refactor: Layout & Plugin for - Demo Frame Embedding

### PR DESCRIPTION
**Pull request by jkodu**
This refactor tackles the below.

- Enhanced the layout of various pages. A quick summary via screenshots is as below.
image
image
image

- A plugin for iframe embedding of a code snippet.
Usage:
image

Open for discussion on what the decorating tags <<EMBED_START>> & <<EMBED_END>> should be named.

This creates an output like so:
image

### CODE REVIEW

File: _layouts/default.html [hunk 1]
```diff
@@ -1,173 +1,193 @@
...
+      </p>
+    </main>
+    
+    <script src="{{ site.baseurl }}/assets/js/jquery.min.js"></script>
+    <script src="{{ site.baseurl }}/assets/js/bootstrap.min.js"></script>
+    <script>
+      $(document).ready(function () {
+        // append class names to table
+        $('table').each(function() {
+          $(this).addClass('table table-striped table-hover');
```
Comment (WilcoFiers) [inline _layouts/default.html]: This seems like a problem that could be solved without JS, no?

File: _layouts/default.html [hunk 2]
```diff
@@ -1,173 +1,193 @@
...
-        About Us
-      </a></li>
-      <li><a href="{{ site.baseurl }}/pages/contribute.html">
-        How to contribute
-      </a></li>
-      <li><a href="https://www.w3.org/community/auto-wcag/about-us/">GitHub repository</a></li>
-    </ul>
-  </div>
+        // find every language-html and iframe next to eachother and flex them in 2 column layout
+        $('div.language-html')
```
Comment (WilcoFiers) [inline _layouts/default.html]: This too seems like it should work without JS.

File: assets/css/main.scss [hunk 3]
```diff
@@ -2,51 +2,171 @@
...
-    color: #FFF;
-    text-decoration:underline;
-  }
-  ul {
-    padding-left:1em;
-  }
-}
\ No newline at end of file
+
+// table {
```
Comment (WilcoFiers) [inline assets/css/main.scss]: Either this shouldn't be commented out, or this shouldn't be here at all :)

File: development.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+- bundle install (once, if you do not have ruby, install via home brew).
+- bundle exec jekyll serve
+
+This will serve the site at - http://127.0.0.1:4000/auto-wcag/
+
+All changes are watched and served.
+
+
+<!-- To document
```
Comment (WilcoFiers) [inline development.md]: ??

File: spec/frame_embed_generator_spec.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+require 'rspec'  
+require 'rack/test'
+
+set :environment, :test
+
+describe 'FrameEmbedGenerator Plugin Testing' do
+
+    include Rack::Test::Methods
+  
+    # TODO: Test
```
Comment (WilcoFiers) [inline spec/frame_embed_generator_spec.rb]: Should probably change that :-)

File: Guardfile [hunk 6]
```diff
@@ -0,0 +1,73 @@
+# A sample Guardfile
```
Comment (WilcoFiers) [inline Guardfile]: So this is a watcher for frame_embed_generator.rb? Please document the function of this file.

Comment (jkodu) [reply Guardfile]: I have removed Guard, and watching using foreman for spec files. As I still have not learnt how to write tests especially for Jekyll Plugins.

On the contrary to writing rspec for a simple functionality like so - https://github.com/auto-wcag/auto-wcag-deployer/blob/master/spec/server_spec.rb

File: _layouts/default.html (hunk 1 above)
Comment (jkodu) [reply _layouts/default.html]: Yes, can do whilst listening to the Liquid Template Generation, but it is an unnecessary plugin to author. 

I will see what I can do about this.
Will create an issue to solve in the coming days.

File: assets/css/main.scss (hunk 3 above)
Comment (jkodu) [reply assets/css/main.scss]: Removed commented styles.

File: development.md (hunk 4 above)
Comment (jkodu) [reply development.md]: removed...

File: spec/frame_embed_generator_spec.rb (hunk 5 above)
Comment (jkodu) [reply spec/frame_embed_generator_spec.rb]: removed spec file, for reason explained in earlier comment.

File: _layouts/default.html (hunk 1 above)
Comment (jkodu) [reply _layouts/default.html]: Have to master overriding Liquid Template Generation, will give it a stab once I learn more on that.

File: _layouts/default.html (hunk 2 above)
Comment (jkodu) [reply _layouts/default.html]: Same comment as above. Need to master Liquid Template generation.

**Comment by kasperisager** [comment]
Going over this again, I do have some concerns regarding the complexity introduced. We're effectively going from four well-documented and established languages used for authoring pages (YAML, Liquid, Markdown, and HTML) to additionally bolting on our own custom language when we want to do things that we should be able to accomplish in the other languages. On top of that maintainers now also have to know Ruby if they want to touch the custom language that we invent. I see a couple good ways to solve this without introducing too much complexity:

1. Use a templating language that supports components/macros defined in that same templating language.
2. Use web components on the client instead of custom syntax as part of the build process.

All that being said, I still think the end result is great and hat tip to @jkodu for your work on this!

**Comment by jkodu** [comment]
@WilcoFiers 
@kasperisager 

Magic! Got there.
Solved for all the above arguments in commit - https://github.com/auto-wcag/auto-wcag/commit/f95b006b3943115cc292c37c41df4ef486d8beda

This solves the enhancement issue - https://github.com/auto-wcag/auto-wcag/issues/98

No overhead in terms of writing rules, the snippets look like below:
image

The output is as so:
image

**Comment by jkodu** [comment]
@WilcoFiers @skotkjerra @Brynanders 
Need a few more approvals on this to merge.

**Pull request merged**